src/backend/src/services/teams.services.ts

```diff
@@ -284,4 +284,38 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  /**
+   * @param submitter a user who's archiving the team
+   * @param teamId a id of team to be updated
+   * @returns the archived team
+   * @throws if the team is not found, the submitter has no privilege, the team has any projects that are not complete
+   */
+  static async archiveTeam(subimtter: User, teamId: string): Promise<Team> {
+    const team = await prisma.team.findFirst({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
```

Comment:
should also throw an error if it's already been deleted

Reply:
We hard delete teams so this should be covered by not finding the team